core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -151,15 +151,15 @@ private <T> T execute(
 
     String fullUri = String.format("%s/%s", uri, path);
     HttpUriRequestBase request = new HttpUriRequestBase(method.name(), URI.create(fullUri));
-    addRequestHeaders(request);
 
-    if (requestBody != null) {
-      try {
-        StringEntity stringEntity = new StringEntity(mapper.writeValueAsString(requestBody));
-        request.setEntity(stringEntity);
-      } catch (JsonProcessingException e) {
-        throw new RESTException(e, "Failed to write request body: %s", requestBody);
-      }
+    if (requestBody instanceof Map) {
```

Comment:
This change updates `HTTPClient` so that a string map will be form encoded rather than JSON encoded. This minimizes the changes needed to support requests to the `v1/oauth/token` endpoint, but an alternative is to add another method, `RESTClient.formPost`, to send map data encoded as `application/x-www-form-urlencoded` instead of JSON.

@danielcweeks, what do you think?

Reply:
Actually, the request body in `post` must be a `RESTRequest`, so we'll have to add the `formPost` method. I think that's probably the right solution anyway.

Reply:
Added `postForm` for this.